### ISSUE TITLE
Remove media player controls from quick settings

### ISSUE THREAD

**Issue by rien333** [siavash79/AOSPMods#546]
## What's on your mind?

Numerous apps (vlc, spotify, kore, netflix, doubletwist) display a "media player notification" in the quick settings panel (see screenshot below).  As lots of users have reported, _there is no way to remove this notification_, despite settings that would, seemingly, imply the contrary (see e.g. [reddit](https://www.reddit.com/r/GooglePixel/comments/x2ktf9/media_players_remains_in_the_extended/), [xda](https://forum.xda-developers.com/t/how-to-disable-the-media-output-notification.4409325/)). That is, the settings under `Sound & vibration > Media` simply do not work. Nor does disabling, say, all vlc notifications, or simply swiping the notification away.

Thus, my request is to allow AOSPmods users to permanently disable this notification. 

Implementation-wise, maybe it's helpful to check what the settings under `Sounds & vibrations > Media` are actually doing?

## Is this feature implemented on any other app/mod/rom? please specify

Doesn't seem like it. There's #55, but the discussion is so sparse that I barely understand what it is about, or why the issue was closed.

### Screenshot

![Screenshot_20230312-123949_Firefox](https://user-images.githubusercontent.com/10417027/224542970-f6817829-04e5-44d3-b6cb-0309f8a7a944.png)

**Comment by EleoXDA** [comment]
No. #55 is not related to what you ask. Currently in android any audio steaming app should show some kind of control in qs. My question is why do you want to remove it? It is a feature, not a bug

**Comment by rien333** [comment]
> My question is why do you want to remove it? It is a feature, not a bug

It's clutter. If you look at the Reddit post I linked, and [Google](https://www.google.com/search?q=android+root+remove+media+notification+from+quick+settings+&source=hp&oq=android+root+remove+media+notification+from+quick+settings+&gs_l=mobile-heirloom-hp.12...1606.25308.0.25798.68.60.4.4.4.0.222.5541.44j15j1.60.0....0...1c.1j4.34.mobile-heirloom-hp..33.35.2690.IxqunoxjLFU)   around for yourself, you will see tons and tons of people that want to get rid of it, but can't. That means the demand is there.

I think some reasons for wanting it gone are:

1. Minimize visual distractions
2. Get more screen real estate for notifications/stuff you actually want to see
3. Save on some processing power (presumably, the less things there are to draw, the snappier everything will be)
4. Avoid unwanted inputs 
5. A desire to control one's operating system, and to remove duplicate functionality (I already have plenty of ways to control media)

Also, I think calling it feature may be a bit too quick - a feature is only a feature if you intend on using it. In any case, its weird to have an OS that prioritizes customization, but doesn't let you customize a pretty big part of its appearance. Call it OCD, but something like that defintely seems in line with the scope of this project.

**Comment by EleoXDA** [comment]
> It's clutter. If you look at the Reddit post I linked, and [Google](https://www.google.com/search?q=android+root+remove+media+notification+from+quick+settings+&source=hp&oq=android+root+remove+media+notification+from+quick+settings+&gs_l=mobile-heirloom-hp.12...1606.25308.0.25798.68.60.4.4.4.0.222.5541.44j15j1.60.0....0...1c.1j4.34.mobile-heirloom-hp..33.35.2690.IxqunoxjLFU)   around for yourself, you will see tons and tons of people that want to get rid of it, but can't. That means the demand is there.
> 

I can see why people want to get more control over what is shown. Statistically you can find a person wanting exactly all the same stuff as you ever wanted. The fact that users want to get rid of it doesn't mean there is a high demand.


> I think some reasons for wanting it gone are:
> 
> 1. Minimize visual distractions

It is not distractions. Instead you get a control over media stream without openning the media app. It is a feature.

> 2. Get more screen real estate for notifications/stuff you actually want to see

It doesn't take that much space. But lucky that AOSPMods allows you to configure number of qqs toggles, so you can save space in qs that you want to have in notifications 

> 3. Save on some processing power (presumably, the less things there are to draw, the snappier everything will be)

Believe me, it is consuming more power to open the media app every time to control the media stream than having it on qs.

> 4. Avoid unwanted inputs 

That is a user error. You could pretty much shut down the phone by accidentally long pressing power button or clicking the power icon in qs page

> 5. A desire to control one's operating system, and to remove duplicate functionality (I already have plenty of ways to control media)

That is subjective. Objectively, it gives you opportunity to do that in case you don't have it. E.g. you are not using headphones.

> 
> Also, I think calling it feature may be a bit too quick - a feature is only a feature if you intend on using it. In any case, its weird to have an OS that prioritizes customization, but doesn't let you customize a pretty big part of its appearance. Call it OCD, but something like that defintely seems in line with the scope of this project.

AndroidOS doesn't prioritize customization. It allows you to do it. Those are two different things. And yes, without AOSPMods you can't still customize a very big part of UI. Which is their decision.

And lastly, no. Minimification is not in scope of AOSPMods. Or UI-simplification. Or having "less clutter".

AOSPMods prioritizes functions. More functions the better. And ability to enable disable those functions. But as said earlier, removing clutter or making UI more user friendly is not in scope of AOSPMods

**Comment by rien333** [comment]
Thanks for your reply. I do believe your reasoning is a bit too quick at times, but I can see you not wanting to invest time in this, especially because you seem to like the feature so much.

> Statistically you can find a person wanting exactly all the same stuff as you ever wanted. The fact that users want to get rid of it doesn't mean there is a high demand.

The point was never that there was "high" demand - I have at best shown that 50-100 people have been bothered by it,  and thought that was roughly sufficient.

In fact, a good reason for _not_ implementing this is that the current behavior may be bug, as evidenced by  the `Sounds & vibrations > Media` settings  doing (next to) nothing. This means the AOSP developers saw it as a "feature", worthy of being disabled, subjective that term apparently is!

> It is not distractions. Instead you get a control over media stream without openning the media app. It is a feature.

Okay, the point I was trying to make was that it is distractive to _some_ people.  Pointing out that it isn't to you is just begging the question. I know that wether something is a distraction is "subjective" - or perhaps better, intersubjective! - but, then again, this app is literally a series of personal choices. 

>  It doesn't take that much space.

It is literally always the biggest UI element in my qs (or maybe phone?). So optimizing that instead of removing qs tiles, as you suggest, might be a more space efficient solution to people. 

> Believe me, it is consuming more power to open the media app every time to control the media stream than having it on qs.

Again, it seems that you greatly benefit from this feature, and that is great. I and other people don't, because we might, say,  always have plenty of other means of controlling media.

> That is a user error. 

It is! How great if we could minimize the chance of that happening! Indeed, relative to the examples you cite, _long_ pressing and seeing the full settings menu, the risk of user error seems much higher here. Then again, this was not the most pressing point.

> And ability to enable disable those functions. But as said earlier, removing clutter or making UI more user friendly is not in scope of AOSPMods

There is plenty of options that remove or substract UI elements, so you can understand my "confusion". There is literally  an option to hide the brightness slider. Are you really saying that all the icon hiding options under "status bar" are different in kind from my suggestion? And so for the ability to set the QS tiles to less than 4?

**Comment by EleoXDA** [comment]
Well, as I said, statistically you can find even more people who would rather remove phone and messaging from android devices

But

Currently the issue is that pixel firmware is undergoing a large update. First QPR was so different that we had to drop support for pre-December firmware. We don't know what new problems will second QPR bring. Currently we are trying to support previous modifications that we cannot do (due to changes in source code).

And

Probably it is as you said: AOSP developers think of it as feature to be disabled - then lets wait patiently for its disappearance. Also maybe it will be even more strengthened (this is my opinion) which will make it even better for usage.

In conclusion,

Removing the only media control notification does NOT look like a good UX (user experience) when I look for a broad range. And this is nothing like adding a new statusbar icon or changing qs toggle row/column numbers.

Thus, in my opinion, **_at least for the time bein_g** I will move it to not planned section. Maybe with time it will be clearer how we should move forward.
